cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java

```diff
@@ -942,6 +924,38 @@ void shouldConvertRegularParamsToFacetParamsWhenTheyMatchAndAreUsingFacetsVersio
         assertThat(actualId.toString(), not(containsString("participant=")));
     }
 
+    @ParameterizedTest
+    @MethodSource("differentQueryParamsProvider")
+    void shouldReturnResultsEvenWhenUsingSpecialCharacters(String param, String processedParam) throws Exception {
+        cristinApiClientStub = spy(cristinApiClientStub);
+        doReturn(cristinApiClientStub).when(clientProvider).getVersionOne();
+        handler = new QueryCristinProjectHandler(clientProvider, new Environment());
+        final var queryParams = Map.of("query", param);
+        final var response = sendQuery(queryParams);
+        var captor = ArgumentCaptor.forClass(URI.class);
+
+        verify(cristinApiClientStub).fetchQueryResults(captor.capture());
+        var actualURI = captor.getValue().toString();
+
+        assertThat(actualURI, containsString(processedParam));
+        assertEquals(HTTP_OK, response.getStatusCode());
+    }
+
+    private static Stream<Arguments> differentQueryParamsProvider() {
+        return Stream.of(
+            Arguments.of("årsstudium",
+                         "årsstudium"),
+            Arguments.of("Et prestisjefylt, stort og viktig prosjekt",
+                         "Et,prestisjefylt,,stort,og,viktig,prosjekt"),
+            Arguments.of("SVIP (styrket veiledning i praksis)",
+                         "SVIP,(styrket,veiledning,i,praksis)"),
+            Arguments.of("Helsefagarbeiderutdanningen – økt gjennomføring",
+                         "Helsefagarbeiderutdanningen,–,økt,gjennomføring"),
+            Arguments.of("<script>alert(1)</script>",
+                         "%3Cscript%3Ealert(1)%3C/script%3E")
+        );
```

Comment:
You could annotate the different test data (for clarity about what is being tested) with Named https://junit.org/junit5/docs/5.9.1/api/org.junit.jupiter.api/org/junit/jupiter/api/Named.html